src/sentry/tasks/relay.py

```diff
@@ -240,12 +307,63 @@ def compute_projectkey_config(key):
         return get_project_config(key.project, project_keys=[key], full_config=True).to_dict()
 
 
+def project_keys_to_update(organization_id=None, project_id=None, public_key=None):
+    """Returns the project keys which need to have their config updated.
+
+    Queries the database for the required project keys.
+    """
+    from sentry.models import Project, ProjectKey
+
+    if organization_id:
+        projects = list(Project.objects.filter(organization_id=organization_id))
+        keys = list(ProjectKey.objects.filter(project__in=projects))
+    elif project_id:
+        projects = [Project.objects.get(id=project_id)]
+        keys = list(ProjectKey.objects.filter(project__in=projects))
+    elif public_key:
+        try:
+            keys = [ProjectKey.objects.get(public_key=public_key)]
+        except ProjectKey.DoesNotExist:
+            # In this particular case, where a project key got deleted and
+            # triggered an update, we know that key doesn't exist and we want to
+            # avoid creating more tasks for it.
+            #
+            # In other similar cases, like an org being deleted, we potentially
+            # cannot find any keys anymore, so we don't know which cache keys
+            # to delete.
+            projectconfig_cache.set_many({public_key: {"disabled": True}})
+            keys = []
```

Comment:
since this function is now only used by the `build_project_config` task my proposal would be to delete the function and place this bit of code directly in the task.  I'd then use the resulting key on `compute_projectkey_config()` and save the result in the cache again directly in the task,  which also allows you to delete `compute_project_configs()`.